sympy/physics/mechanics/actuator.py

```diff
@@ -873,3 +875,82 @@ def __repr__(self):
         else:
             string += ')'
         return string
+
+
+class DuffingSpring(ForceActuator):
+    """A nonlinear spring based on the Duffing equation.
+    Explanation
+    ===========
+    Here, ``DuffingSpring`` represents the force exerted by a nonlinear spring based on the Duffing equation:
+    F = -beta*x-alpha*x**3, where x is the displacement from the equilibrium position, beta is the linear spring constant,
+    and alpha is the coefficient for the nonlinear cubic term.
+    Parameters
+    ==========
+    linear_stiffness : Expr
+        The linear stiffness coefficient (beta).
+    nonlinear_stiffness : Expr
+        The nonlinear stiffness coefficient (alpha).
+    pathway : PathwayBase
+        The pathway that the actuator follows.
+    equilibrium_length : Expr, optional
+        The length at which the spring is in equilibrium (x).
+    """
+
+    def __init__(self, linear_stiffness, nonlinear_stiffness, pathway, equilibrium_length=S.Zero):
+        try:
+            self._linear_stiffness = sympify(linear_stiffness)
+            self._nonlinear_stiffness = sympify(nonlinear_stiffness)
+            self._equilibrium_length = sympify(equilibrium_length)
+        except SympifyError:
+            raise SympifyError("Input arguments could not be sympified.")
+
+        if not isinstance(pathway, PathwayBase):
+            raise TypeError("pathway must be an instance of PathwayBase.")
+        self._pathway = pathway
+
+    @property
+    def linear_stiffness(self):
+        return self._linear_stiffness
+
+    @linear_stiffness.setter
+    def linear_stiffness(self, value):
+        raise AttributeError("Cannot modify linear_stiffness")
```

Comment:
If you only specify a getter of a property, trying to set a value will already raise an AttributeError. Therefore, you can remove the setters that only raise an AttributeError, i.e.:
```python
def __init__(self, ...):
    self._linear_stiffness = sympify(linear_stiffness, strict=True)
    ...

@property
def linear_stiffness(self):
    return self._linear_stiffness
```
If you prefer having a separate setter to match the implementation of `LinearSpring`, `LinearDamper`, etc., then I would propose trying to use the same format, i.e.:
```python
def __init__(self, ...):
    self.linear_stiffness = linear_stiffness
    ...

@property
def linear_stiffness(self):
    return self._linear_stiffness

@linear_stiffness.setter
def linear_stiffness(self, linear_stiffness):
        if hasattr(self, '_linear_stiffness'):
            msg = (
                f'Can\'t set attribute `linear_stiffness` to '
                f'{repr(linear_stiffness)} as it is immutable.'
            )
            raise AttributeError(msg)
        self._linear_stiffness = sympify(linear_stiffness, strict=True)
```